gamemode/modules/base/cl_util.lua

```diff
@@ -21,15 +21,16 @@ usermessage.Hook("blackScreen", blackScreen)
 --[[---------------------------------------------------------------------------
 Wrap strings to not become wider than the given amount of pixels
 ---------------------------------------------------------------------------]]
-local function charWrap(text, pxWidth)
+local function charWrap(text, pxWidth, fullWidth)
     local total = 0
 
     text = text:gsub(".", function(char)
         total = total + surface.GetTextSize(char)
 
         -- Wrap around when the max width is reached
         if total >= pxWidth then
-            total = 0
+            total = surface.GetTextSize(char) -- total needs to include the character width because it's inserted in a new line
+            pxWidth = fullWidth
```

Comment:
This looks off, the `pxWidth` argument of this function is different than the `pxWidth` variable in `DarkRP.textWrap`, which is given as the third argument here. That's confusing. Also, the `pxWidth` argument seems to be used only in the first iteration of this function, only to be replaced by `fullWidth` at the second character. That seems to imply the calculation `pxWidth - (total - wordlen)` given as argument below is disregarded almost always.

Reply:
`fullWidth` does not get replaced at the second character, it gets replaced when we get into a new line. Both of them could be disregarded.